OrbitGl/ProcessManager.cpp

```diff
@@ -95,17 +103,23 @@ void ProcessListManagerImpl::WorkerFunction() {
       continue;
     }
 
-    std::vector<ProcessInfo> processes(response.processes().begin(),
-                                       response.processes().end());
+    std::vector<ProcessInfo> processes;
 
-    absl::MutexLock callback_lock(&callback_mutex_);
-    callback_(std::move(processes));
+    absl::MutexLock callback_lock(&mutex_);
+    process_list_.assign(response.processes().begin(),
+                         response.processes().end());
```

Comment:
I know you have just refactored the code here, but since you touched the line:

I think this `responses.processes().begin/end` can be dangerous when a temporary is returned (and not a reference like in this case.). If gRPC changes in the future or somebody changes the type of `response` this will silently fail and probably crash. Putting the result of `response.processes()` in a const-ref variable won't hurt performance but will resolve this potential issue.

Reply:
good point - done.